src/main/java/org/jabref/gui/preferences/GeneralTabViewModel.java

```diff
@@ -0,0 +1,186 @@
+                    Localization.lang("You have changed the language setting.")
+                                .concat(" ")
+                                .concat(Localization.lang("You must restart JabRef for this to come into effect.")));
+        }
+        preferences.setDefaultEncoding(selectedEncodingProperty.getValue());
+        preferences.putBoolean(JabRefPreferences.BIBLATEX_DEFAULT_MODE, selectedBiblatexModeProperty.getValue() == BibDatabaseMode.BIBLATEX);
+
+        preferences.putBoolean(JabRefPreferences.WARN_ABOUT_DUPLICATES_IN_INSPECTION, inspectionWarningDuplicateProperty.getValue());
+        preferences.putBoolean(JabRefPreferences.CONFIRM_DELETE, confirmDeleteProperty.getValue());
+        preferences.putBoolean(JabRefPreferences.ENFORCE_LEGAL_BIBTEX_KEY, enforceLegalKeysProperty.getValue());
+        if (preferences.getBoolean(JabRefPreferences.MEMORY_STICK_MODE) && !memoryStickModeProperty.getValue()) {
+            dialogService.showInformationDialogAndWait(Localization.lang("Memory stick mode"),
+                    Localization.lang("To disable the memory stick mode"
+                            + " rename or remove the jabref.xml file in the same folder as JabRef."));
+        }
+        preferences.putBoolean(JabRefPreferences.MEMORY_STICK_MODE, memoryStickModeProperty.getValue());
+        preferences.setShouldCollectTelemetry(collectTelemetryProperty.getValue());
+        preferences.putBoolean(JabRefPreferences.SHOW_ADVANCED_HINTS, showAdvancedHintsProperty.getValue());
+
+        preferences.putBoolean(JabRefPreferences.USE_OWNER, markOwnerOverwriteProperty.getValue());
+        preferences.put(JabRefPreferences.DEFAULT_OWNER, markOwnerNameProperty.getValue().trim());
+        preferences.putBoolean(JabRefPreferences.OVERWRITE_OWNER, markOwnerOverwriteProperty.getValue());
+
+        preferences.putBoolean(JabRefPreferences.USE_TIME_STAMP, markTimestampProperty.getValue());
+        preferences.put(JabRefPreferences.TIME_STAMP_FORMAT, markTimeStampFormatProperty.getValue().trim());
+        preferences.put(JabRefPreferences.TIME_STAMP_FIELD, markTimeStampFieldNameProperty.getValue().trim());
+        preferences.putBoolean(JabRefPreferences.OVERWRITE_TIME_STAMP, markTimeStampOverwriteProperty.getValue());
+        preferences.putBoolean(JabRefPreferences.UPDATE_TIMESTAMP, updateTimeStampProperty.getValue());
+
+        // Update name of the time stamp field based on preferences
+        InternalBibtexFields.updateTimeStampField(preferences.get(JabRefPreferences.TIME_STAMP_FIELD));
+    }
+
+    public boolean validateSettings() {
+        try {
+            // Test if date format is legal:
+            DateTimeFormatter.ofPattern(markTimeStampFormatProperty.getValue());
+        } catch (IllegalArgumentException exception) {
+            DefaultTaskExecutor.runInJavaFXThread(() -> dialogService.showErrorDialogAndWait(Localization.lang("Invalid date format"),
```

Comment:
you should already be on the FX thread or is this method called fom a non fx thread`?

Reply:
This will be gone as soon as the validators are implemented